src/h5web/visualizations/containers/NxSpectrumContainer.tsx

```diff
@@ -1,39 +1,58 @@
 import React, { ReactElement, useState, useContext } from 'react';
 import { range } from 'lodash-es';
-import { HDF5SimpleShape } from '../../providers/models';
-import { useDatasetValue } from './hooks';
-import { assertGroup } from '../../providers/utils';
+import { HDF5SimpleShape, HDF5Dataset } from '../../providers/models';
+import { useDatasetValues } from './hooks';
+import { assertGroup, isDataset } from '../../providers/utils';
 import DimensionMapper from '../../dimension-mapper/DimensionMapper';
 import { DimensionMapping } from '../../dimension-mapper/models';
 import MappedLineVis from '../line/MappedLineVis';
 import { ProviderContext } from '../../providers/context';
-import { getSignalDataset } from '../nexus/utils';
+import {
+  getSignalDataset,
+  getLinkedEntity,
+  getAxesLabels,
+} from '../nexus/utils';
 import { VisContainerProps } from './models';
 
 function NxSpectrumContainer(props: VisContainerProps): ReactElement {
   const { entity } = props;
   assertGroup(entity);
 
   const { metadata } = useContext(ProviderContext);
-  const dataset = getSignalDataset(entity, metadata);
+  const signalDataset = getSignalDataset(entity, metadata);
 
-  if (!dataset) {
+  if (!signalDataset) {
     throw new Error('Signal dataset not found');
   }
 
-  const { dims } = dataset.shape as HDF5SimpleShape;
+  const { dims } = signalDataset.shape as HDF5SimpleShape;
   if (dims.length < 1) {
     throw new Error('Expected dataset with at least one dimension');
   }
 
+  const axesLabels =
+    getAxesLabels(entity) || Array(dims.length).fill(undefined);
+  if (dims.length !== axesLabels.length) {
+    throw new Error(
+      `Number of axes ${axesLabels.length} does not correspond to the dataset dimensions (${dims})`
+    );
+  }
```

Comment:
If I'm not mistaken, the rest of the code still works if `axesLabels` has fewer elements than the number of dimensions. So we could just support this use case and assume that `['X']` for a 2D dataset is equivalent to `['X', '.']`. This would allow the fallback of `axesLabels` to just be `[]` instead of an array of `undefined` values.

Reply:
The code should work but the NeXuS specification does not allow it (https://manual.nexusformat.org/classes/base_classes/NXdata.html). Should I do as you say and leave the verification to #254 ?

Reply:
Well, the goal of #254 is not to validate the file but to help users debug invalid syntax that h5web does not support. In this case it is easier to support the invalid syntax than to validate it, so I would argue that being permissive makes sense.

Reply:
I don't know. 

It seems dangerous to me to assume that `['X']` should be `['X', '.']` (why not `['.', 'X']`) ? We may end up attributing the wrong dataset to the wrong dimensions leading to error in `LineVis` while the source is related to NeXuS.